crates/ra_parser/src/grammar/paths.rs

```diff
@@ -85,6 +87,12 @@ fn path_segment(p: &mut Parser, mode: Mode, first: bool) {
             // use crate::foo;
             T![self] | T![super] | T![crate] => p.bump_any(),
             _ => {
+                if empty {
+                    // test empty_segment
```

Comment:
I think this should be `test_err`

Reply:
Right, it works becase we don't produce an error message here, but we should do this. It's a bug that it works with test and not test_err

Reply:
Thing is, it fails with `test_err` because we don't produce an error. I see that `err_recover` calls `m.complete` so I'm not sure...

Reply:
You don't produce error because you `return` before producing an error :-) I think you can just move the `if` after the next line?

Reply:
But `err_recover` calls `m.complete`, so we can't call `abandon`? Or can we?

Reply:
`err_recover` doesn't use `m` at all?

Reply:
I was looking at the following snippet from that method:

```rust
        let m = self.start();
        self.error(message);
        self.bump_any();
        m.complete(self, ERROR);
```

But it works, I guess, with a parsing difference in one of the existing tests.